tests/test_interpolation.py

```diff
@@ -590,3 +590,85 @@ def test_msf_interpolate():
     assert np.all(np.unique(nzt) == np.array([1, 2, 3, 4]))
     # 12 points x 4 timesteps
     assert nzt.size == 48
+
+
+@pytest.mark.parametrize('shape', [(50, 50, 50)])
+@pytest.mark.parametrize('so', (2, 4, 8))
+@pytest.mark.parametrize('tn', (40, 40, 200))
+def test_decompose_to_aligned(shape, so, tn):
```

Comment:
is this really an MFE? It's huge

Reply:
It is not an MFE for nested source injection, but has just the necessary extra stuff to test nested source injection + listing 3 of the paper. So it is more than an MFE but still all of it seems useful to my eyes.